pontoon/machinery/views.py

```diff
@@ -134,7 +134,10 @@ def microsoft_translator(request):
         )
 
     # Validate if locale exists in the database to avoid any potential XSS attacks.
-    if not Locale.objects.filter(ms_translator_code=locale_code).exists():
+    if (
+        not locale_code
+        or not Locale.objects.filter(ms_translator_code=locale_code).exists()
+    ):
```

Comment:
Now that we only allow these requests for logged-in users, this query lost its purpose. I think we should just remove them.